src/prefect/backend/flow_run.py

```diff
@@ -0,0 +1,596 @@
+            Storage. If `None`, the flow's Storage metadata will be pulled from the
+            API and used to get a functional instance of the Flow and its tasks.
+        runner_cls: An optional `FlowRunner` to override the default `CloudFlowRunner`
+        **kwargs: Additional kwargs will be passed to the `FlowRunner.run` method
+
+    Returns:
+        A `FlowRun` instance with information about the state of the flow run and its
+        task runs
+    """
+    logger.debug(f"Querying for flow run {flow_run_id!r}")
+
+    # Get the `FlowRunner` class type
+    # TODO: Respect a config option for this class so it can be overridden by env var,
+    #       create a separate config argument for flow runs executed with the backend
+    runner_cls = runner_cls or prefect.engine.cloud.flow_runner.CloudFlowRunner
+
+    # Get a `FlowRun` object
+    flow_run = FlowRun.from_flow_run_id(
+        flow_run_id=flow_run_id, load_static_tasks=False
+    )
+
+    logger.info(f"Constructing execution environment for flow run {flow_run_id!r}")
+
+    # Populate global secrets
+    secrets = prefect.context.get("secrets", {})
+    if flow_run.flow.storage:
+        logger.info(f"Loading secrets...")
+        for secret in flow_run.flow.storage.secrets:
+            with fail_flow_run_on_exception(
+                flow_run_id=flow_run_id,
+                message=f"Failed to load flow secret {secret!r}: {{exc}}",
+            ):
+                secrets[secret] = prefect.tasks.secrets.PrefectSecret(name=secret).run()
+
+    # Load the flow from storage if not explicitly provided
+    if not flow:
+        logger.info(f"Loading flow from {flow_run.flow.storage}...")
+        with prefect.context(secrets=secrets, loading_flow=True):
+            with fail_flow_run_on_exception(
+                flow_run_id=flow_run_id,
+                message="Failed to load flow from storage: {exc}",
+            ):
+                flow = flow_run.flow.storage.get_flow(flow_run.flow.name)
+
+    # Update the run context to include secrets with merging
+    run_kwargs = copy.deepcopy(kwargs)
+    run_kwargs["context"] = run_kwargs.get("context", {})
+    run_kwargs["context"]["secrets"] = {
+        # User provided secrets will override secrets we pulled from storage and the
+        # current context
+        **secrets,
+        **run_kwargs["context"].get("secrets", {}),
+    }
+    # Update some default run kwargs with flow settings
+    run_kwargs.setdefault("executor", flow.executor)
+
+    # Execute the flow, this call will block until exit
+    logger.info(
+        f"Beginning execution of flow run {flow_run.name!r} from {flow_run.flow.name!r} "
+        f"with {runner_cls.__name__!r}"
+    )
+    with prefect.context(flow_run_id=flow_run_id):
+        with fail_flow_run_on_exception(
+            flow_run_id=flow_run_id,
+            message="Failed to execute flow: {exc}",
+        ):
+            if flow_run.flow.run_config is not None:
```

Comment:
I believe `run_config` can be None in cases in which we don't want to use `flow.environment`

Reply:
This logic is straight from `prefect execute flow-run` so I don't think this is true

Reply:
I'm probably full of it then